### ISSUE TITLE
more than two snapshots

### ISSUE THREAD

**Issue by cruzchue** [maziarraissi/PINNs#32]
The code for the Korteweg–de Vries problem (KdV.py) provides an example using only two time snapshots.  I am wondering which parts need to be edited/added in order to use three or more time snapshots.

**Comment by nish-ant** [comment]
As proposed in the paper, the neural network, by design, is trained by minimizing the loss function which is a sum of squared errors calculated at **two** distinct time steps (equation (17) in [paper](https://arxiv.org/pdf/1711.10566.pdf)).

For including information from more than two time steps, the calculation of [loss](https://github.com/maziarraissi/PINNs/blob/0542794b0a91b9e8764a38f5fc9cd9647a3929ba/main/discrete_time_identification%20(KdV)/KdV.py#L67) can be done such that it iterates over all the snapshots.

**Comment by cruzchue** [comment]
Thanks for the quick response.  The part of the loss function is certainly clear and well described along the four example. I still have some related questions.  First, I describe my understanding and the problem, then I post the questions.

** Problem **

For calculating the loss, the script first computes the nonlinear operators in the functions [net_U0](https://github.com/maziarraissi/PINNs/blob/0542794b0a91b9e8764a38f5fc9cd9647a3929ba/main/discrete_time_identification%20(KdV)/KdV.py#L122) and [net_U1](https://github.com/maziarraissi/PINNs/blob/0542794b0a91b9e8764a38f5fc9cd9647a3929ba/main/discrete_time_identification%20(KdV)/KdV.py#L133).  That is, net_U0 is for time snapshot 0, and net_U1 is for time snapshot 1: two time snapshots.

Now, for a third time snapshot, we would require a net_U2, which should not be a problem.  But in both net_U0 and net_U1, there is a time step variable called "dt", which is the time difference between only two snapshot.

**Questions:**

- For a three-snapshot problem, should I define two time steps (dt1, dt2) instead of one (dt)?

- In that case, both dt1 and dt1 should be time step with respect to the first snapshot (t1-t0, t2-t0)?, or should they be difference among sequential times (t1-t0, t2-t1)?

**Comment by nish-ant** [comment]
Alright, now I understand the question better.
I don't think the way you're trying to make it work will be possible in the given framework.

The time integration method is based on Runge-Kutta method, which takes intermediate steps between two time steps.
The implicit RK [equations](https://en.wikipedia.org/wiki/Runge%E2%80%93Kutta_methods#Implicit_Runge%E2%80%93Kutta_methods) show this clearly.with the order ranging between q and 2q (q = number of stages).

However, a [multistep method](https://arxiv.org/pdf/1801.01236.pdf) might be better suited for your task.

**Comment by cruzchue** [comment]
Many thanks for the information.  For sure, we will take a look at that.
